src/map/skill.cpp

```diff
@@ -13724,6 +13724,16 @@ static int8 skill_castend_id_check(struct block_list *src, struct block_list *ta
 					return USESKILL_FAIL_MAX;
 			}
 			break;
+		case DC_WINKCHARM:
+			{
+				status_data* tstatus = status_get_status_data(*target);
+				if (tstatus->race == RC_DEMON || tstatus->race == RC_DEMIHUMAN || tstatus->race == RC_ANGEL)
+					break;
+				if (tstatus->race == RC_PLAYER_HUMAN || tstatus->race == RC_PLAYER_DORAM)
+					break;
+				return USESKILL_FAIL_LEVEL;
+			}
```

Comment:
```suggestion
				switch (status_get_status_data(*target)->race) {
				case RC_DEMON:
				case RC_DEMIHUMAN:
				case RC_ANGEL:
				case RC_PLAYER_HUMAN:
				case RC_PLAYER_DORAM:
					break;
				default:
					return USESKILL_FAIL_LEVEL;
				}
```
or `switch (status_data* tstatus = status_get_status_data(*target); tstatus->race) {` if you need tstatus in the switch

Reply:
Yeah I also like the switch more